src/vs/workbench/services/authentication/browser/authenticationService.ts

```diff
@@ -716,7 +716,16 @@ export class AuthenticationService extends Disposable implements IAuthentication
 		try {
 			const authProvider = this._authenticationProviders.get(id) || await this.tryActivateProvider(id, activateImmediate);
 			return await authProvider.getSessions(scopes);
-		} catch (_) {
+		} catch (e) {
+			if (e.message === 'Cancelled' || e === 'Cancelled') {
+				throw e;
+			}
+
+			// thrown on browser cancellation by microsoft-authentication provider
+			if (e === 'access_denied') {
+				throw new Error('Cancelled');
+			}
```

Comment:
Please don't hate me for this @olegoid (getting familiar as the new owner of this code 😆 ) but this code change here has convinced me that you had it right the first time...

The reason I'm changing my mind is because we, ideally, wanna give Auth Providers the ability to expose their own (potentially context aware) error messages.

Sorry for flip flopping on you, but you've made me think of the problem a bit differently so thank you for that!

Reply:
Heh, no worries! Just reverted the PR to what it was a day ago.